src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapperTest.java

```diff
@@ -1,19 +1,217 @@
+@ExtendWith(MockitoExtension.class)
 class DiagnosticReportMapperTest {
 
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @Mock
+    private ResourceFullUrlGenerator resourceFullUrlGenerator;
+
+    @InjectMocks
+    private DiagnosticReportMapper mapper;
+
+    @Test
+    void testMapToDiagnosticReportMissingRequiredSegments() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06" // InvestigationSubject
+        ));
+
+        assertThatThrownBy(() -> mapper.mapToDiagnosticReport(message, null, null, null, null, null))
+            .isExactlyInstanceOf(MissingSegmentException.class)
+            .hasMessageStartingWith("EDIFACT section is missing segment");
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithMessageWithoutResourceReferences() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var result = mapper.mapToDiagnosticReport(message, null, null, null, null, null);
+        ZonedDateTime dateIssued = LocalDateTime.of(2010, 02, 25, 15, 41).atZone(TimestampService.UK_ZONE);
+
+        assertAll(
+            () -> assertThat(result.getIdentifier())
+                .first()
+                .satisfies(identifier -> assertAll(
+                    () -> assertThat(identifier.getValue()).isEqualTo("15/CH000042P/200010191704")
+                )),
+            () -> assertThat(result.getCode().getCoding())
+                .hasSize(1)
+                .first()
+                .satisfies(coding -> assertAll(
+                    () -> assertThat(coding.getCode()).isEqualTo("721981007"),
+                    () -> assertThat(coding.getDisplay()).isEqualTo("Diagnostic studies report"),
+                    () -> assertThat(coding.getSystem()).isEqualTo("http://snomed.info/sct")
+                )),
+            () -> assertThat(result.getStatus()).isEqualTo(DiagnosticReport.DiagnosticReportStatus.UNKNOWN),
+            () -> assertThat(result.getIssued()).isEqualTo(Date.from(dateIssued.toInstant()))
+        );
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithSubjectResource() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var patient = mock(Patient.class);
+        when(resourceFullUrlGenerator.generate(any(Patient.class))).thenReturn("patient-full-url");
+
+        final var result = mapper.mapToDiagnosticReport(message, patient, null, null, null, null);
+
+        assertAll(
+            () -> assertThat(result.getSubject())
+                .extracting(Reference::getReference)
+                .isEqualTo("patient-full-url")
+        );
```

Comment:
No need to assertAll of one assertion
```suggestion
        assertThat(result.getSubject())
            .extracting(Reference::getReference)
            .isEqualTo("patient-full-url");
```

Reply:
Good spot 😄   

Removed it in d61ece4